### DISCUSSION TITLE
SplitButton and DropDownButton : really not recommended ?

### DISCUSSION THREAD

**Discussion by tbolon** [dotnet/winforms#6823] [Q&A] [answered]
I would like to raise a discussion about the #5911 issue, which has been closed by @merriemcgaw, stating that the SplitButton and DropDownButton button style will not be considered because they are deprecated and not recommended anymore.

I was facing a UX problem today which could be resolved with a splitbutton and was questioning myself about the absence of this control on winforms.

After having discovered the [BS_SPLITBUTTON](https://docs.microsoft.com/en-us/windows/win32/controls/button-styles) win32 style I have finally found this issue in the repo, and the reason why this issue was closed.

I also found that the [fluent UI](https://docs.microsoft.com/en-us/windows/apps/design/controls/#alphabetical-index) exposes [multiple variations of the split/dropdown button](https://docs.microsoft.com/en-us/windows/apps/design/controls/buttons) implemented in WinUI.

Bootstrap CSS also implements [split buttons](https://getbootstrap.com/docs/5.1/components/dropdowns/#split-button) and [dropdown buttons](https://getbootstrap.com/docs/5.1/components/dropdowns/#single-button)

I am also sure that a lot of other frameworks implements these controls.

I am sure that there are some accessibility issues with split/dropdown buttons, but I can't find why it must be considered as not recommended? Perhaps I have missed something about that?

I can understand that if the style did not exists in win32, then it should not be developed inside this repo, but since the win32 style exists, I don't really see why it should be ignored?

Thanks,

**Comment by merriemcgaw** [comment, marked as answer]
Do you know how the Win32 [BS_SPLITBUTTON](https://docs.microsoft.com/en-us/windows/win32/controls/button-styles) style button appears on Win11 machines? Many but not all of the Common Controls were updated for a Win11 look and feel, but if this one was not, it would be a lot more work on our end to keep the look and feel consistent on Win11. I'm more than happy to take a proposal from the community if you are able to go through the API Approval logistics and help with the implementation. However, given our current landscape I don't know if the WinForms team will be able to pick up implementing this for .NET 7.

**Reply by tbolon** [reply]
Thank you for your answer,

I just wanted to know why it was considered as not recommended. I see your point now, as the native control has not been updated for win11. I didn't want to ask without having made some search, so I looked globally if this kind of UI was considered a bad practice (which is often the case).

Beside, I have also discovered that not all the required behaviors are implemented natively. It's the responsibility of the developer to implement the dropdown menu.

I hope this discussion could help people coming into this repo asking themselves the same question as me.

Thanks again for your insight.

**Comment by galiprandi** [comment]
If you are new, the [docs home](https://galiprandi.github.io/job-seeker/) has a copyable system prompt to get you started in one paste. No coding experience needed.
